tests/e2e/test-register-ingest.py

```diff
@@ -0,0 +1,158 @@
+PROJECT_NAME = "basic_" + uuid.uuid4().hex.upper()[0:6]
+
+
+@pytest.fixture(scope="module")
+def client(pytestconfig):
+    core_url = pytestconfig.getoption("core_url")
+    serving_url = pytestconfig.getoption("serving_url")
+
+    client = Client(core_url=core_url, serving_url=serving_url,)
+
+    client.set_project(PROJECT_NAME)
+
+    return client
+
+
+@pytest.fixture
+def customer_entity():
+    return Entity(
+        name="customer_id",
+        description="Customer entity for rides",
+        value_type=ValueType.STRING,
+        labels={"team": "customer_service", "common_key": "common_val"},
+    )
+
+
+@pytest.fixture
+def driver_entity():
+    return Entity(
+        name="driver_id",
+        description="Driver entity for car rides",
+        value_type=ValueType.STRING,
+        labels={"team": "matchmaking", "common_key": "common_val"},
+    )
+
+
+@pytest.fixture
+def alltypes_entity():
+    return Entity(
+        name="alltypes_id",
+        description="Driver entity for car rides",
+        value_type=ValueType.STRING,
+        labels={"cat": "alltypes"},
+    )
+
+
+@pytest.fixture
+def alltypes_featuretable():
+    batch_source = DataSource(
+        type="BATCH_FILE",
+        field_mapping={
+            "ride_distance": "ride_distance",
+            "ride_duration": "ride_duration",
+        },
+        options=FileOptions(file_format="parquet", file_url="file://feast/*"),
+        timestamp_column="ts_col",
+        date_partition_column="date_partition_col",
+    )
+    return FeatureTable(
+        name="alltypes",
+        entities=["alltypes_id"],
+        features=[
+            Feature(name="float_feature", dtype=ValueType.FLOAT),
+            Feature(name="int64_feature", dtype=ValueType.INT64),
+            Feature(name="int32_feature", dtype=ValueType.INT32),
+            Feature(name="string_feature", dtype=ValueType.STRING),
+            Feature(name="bytes_feature", dtype=ValueType.BYTES),
+            Feature(name="bool_feature", dtype=ValueType.BOOL),
+            Feature(name="double_feature", dtype=ValueType.DOUBLE),
+            Feature(name="double_list_feature", dtype=ValueType.DOUBLE_LIST),
+            Feature(name="float_list_feature", dtype=ValueType.FLOAT_LIST),
+            Feature(name="int64_list_feature", dtype=ValueType.INT64_LIST),
+            Feature(name="int32_list_feature", dtype=ValueType.INT32_LIST),
+            Feature(name="string_list_feature", dtype=ValueType.STRING_LIST),
+            Feature(name="bytes_list_feature", dtype=ValueType.BYTES_LIST),
+            Feature(name="bool_list_feature", dtype=ValueType.BOOL_LIST),
+        ],
+        max_age=Duration(seconds=3600),
+        batch_source=batch_source,
+        labels={"cat": "alltypes"},
+    )
+
+
+def test_get_list_basic(client: Client, customer_entity: Entity, driver_entity: Entity):
+    basic_ft_spec = FeatureTable.from_yaml(f"{DIR_PATH}/specifications/dev_ft.yaml")
```

Comment:
I think a better test would be `from_yaml` -> register -> get feature table -> to_yaml -> compare yamls

I also think that using `from_yaml` here makes the test a bit brittle. I would prefer that we move all of the test logic into Python code unless we want to test the yaml functionality itself.